internal/v3/translate/translator.go

```diff
@@ -69,16 +63,18 @@ type APIExporter[T any] interface {
 //	  v20250312:
 //
 // In the above case crdVersion is "v1" and majorVersion is "v20250312".
-func NewTranslator(crd *apiextensionsv1.CustomResourceDefinition, crdVersion string, majorVersion string, deps DependencyRepo) *Translator {
-	return &Translator{
-		crd:          CRDInfo{definition: crd, version: crdVersion},
-		majorVersion: MajorVersion{version: majorVersion},
-		deps:         deps,
+// A translator is an immutable configuration object, it can be safely shared
+// across threads
+func NewTranslator(crd *apiextensionsv1.CustomResourceDefinition, crdVersion string, majorVersion string) Translator {
+	return Translator{
+		crd:          &CRDInfo{definition: crd, version: crdVersion},
+		majorVersion: majorVersion,
 	}
 }
 
-// FromAPI translaters a source API structure into a Kubernetes object
-func FromAPI[S any, T any, P PtrClientObj[T]](t *Translator, target P, source *S) ([]client.Object, error) {
+// FromAPI translaters a source API structure into a Kubernetes object.
+// Thread safe: translator is read-only
```

Comment:
general convention from the past: describe the overall thread-safety constraint on the struct level and describe only exceptions on a specific method.